test/1_stdlib/DictionaryLiteral.swift

```diff
@@ -51,8 +51,10 @@ let anNSString = "Foo" as NSString
 var stringNSStringLet: DictionaryLiteral = [ "a": aString as NSString, "b": anNSString]
 expectType(DictionaryLiteral<String, NSString>.self, &stringNSStringLet)
 
+#if false
 var hetero1: DictionaryLiteral = ["a": 1 as NSNumber, "b": "Foo" as NSString]
 expectType(DictionaryLiteral<String, NSObject>.self, &hetero1)
 
 var hetero2: DictionaryLiteral = ["a": 1 as NSNumber, "b": "Foo" as NSString]
 expectType(DictionaryLiteral<String, NSObject>.self, &hetero2)
+#endif
```

Comment:
Yeah, I'm working on a fix independently, which should be an small optimization (prior to this PR) that brings back the functionality above.